### PR TITLE
`tdata` to `teal_data` - `tm_g_swimlane`

### PR DESCRIPTION
**Pull request by vedhav**
**Example App**

```r
data <- teal.data::cdisc_data() |>
  within(library(dplyr)) |>
  within(library(nestcolor)) |>
  within(ADSL <- osprey::rADSL %>%
    dplyr::mutate(TRTDURD = as.integer(TRTEDTM - TRTSDTM) + 1) %>%
    dplyr::filter(STRATA1 == "A" & ARMCD == "ARM A")) |>
  within(ADRS <- osprey::rADRS) |>
  within(ADRS <- ADRS %>%
    dplyr::filter(PARAMCD == "LSTASDI" & DCSREAS == "Death") %>%
    mutate(AVALC = DCSREAS, ADY = EOSDY) %>%
    base::rbind(ADRS %>% dplyr::filter(PARAMCD == "OVRINV" & AVALC != "NE")) %>%
    arrange(USUBJID))

teal.data::datanames(data) <- c("ADSL", "ADRS")
teal.data::join_keys(data) <- teal.data::default_cdisc_join_keys[teal.data::datanames(data)]

ADSL <- data[["ADSL"]]
ADRS <- data[["ADRS"]]

app <- teal::init(
  data = data,
  modules = teal::modules(
    tm_g_swimlane(
      label = "Swimlane Plot",
      dataname = "ADRS",
      bar_var = teal.transform::choices_selected(
        selected = "TRTDURD",
        choices = c("TRTDURD", "EOSDY")
      ),
      bar_color_var = teal.transform::choices_selected(
        selected = "EOSSTT",
        choices = c("EOSSTT", "ARM", "ARMCD", "ACTARM", "ACTARMCD", "SEX")
      ),
      sort_var = teal.transform::choices_selected(
        selected = "ACTARMCD",
        choices = c("USUBJID", "SITEID", "ACTARMCD", "TRTDURD")
      ),
      marker_pos_var = teal.transform::choices_selected(
        selected = "ADY",
        choices = c("ADY")
      ),
      marker_shape_var = teal.transform::choices_selected(
        selected = "AVALC",
        c("AVALC", "AVISIT")
      ),
      marker_shape_opt = c("CR" = 16, "PR" = 17, "SD" = 18, "PD" = 15, "Death" = 8),
      marker_color_var = teal.transform::choices_selected(
        selected = "AVALC",
        choices = c("AVALC", "AVISIT")
      ),
      marker_color_opt = c(
        "CR" = "green", "PR" = "blue", "SD" = "goldenrod",
        "PD" = "red", "Death" = "black"
      ),
      vref_line = c(30, 60),
      anno_txt_var = teal.transform::choices_selected(
        selected = c("ACTARM", "SEX"),
        choices = c(
          "ARM", "ARMCD", "ACTARM", "ACTARMCD", "AGEGR1",
          "SEX", "RACE", "COUNTRY", "DCSREAS", "DCSREASP"
        )
      )
    )
  )
)
if (interactive()) {
  shinyApp(app$ui, app$server)
}
```

### CODE REVIEW

File: R/tm_g_swimlane.R [hunk 1]
```diff
@@ -325,16 +326,16 @@ srv_g_swimlane <- function(id,
     output_q <- reactive({
       teal::validate_inputs(iv())
 
-      validate(need("ADSL" %in% names(data), "'ADSL' not included in data"))
+      validate(need("ADSL" %in% datanames(data()), "'ADSL' not included in data"))
       validate(need(
-        (length(data) == 1 && dataname == "ADSL") ||
-          (length(data) >= 2 && dataname != "ADSL"), paste(
+        (length(datanames(data())) == 1 && dataname == "ADSL") ||
+          (length(datanames(data())) >= 2 && dataname != "ADSL"), paste(
```
Comment (vedhav) [inline R/tm_g_swimlane.R:332]: Creating a `length` method for `teal_data` overkill?

Comment (kartikeyakirar) [reply R/tm_g_swimlane.R:332]: I am not sure if its used in too many places.

**Review by kartikeyakirar — APPROVED** [review]
LGTM👍

File: R/tm_g_swimlane.R (hunk 1 above)
Comment (chlebowa) [reply R/tm_g_swimlane.R:332]: Not overkill but wrong. What would the length represent? There are several features to a `teal_data` beyond `datanames`.

Comment (vedhav) [reply R/tm_g_swimlane.R:332]: Makes sense. Too ambiguous.

**Pull request merged**